clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -5594,7 +5594,11 @@ void CodeGenModule::EmitGlobalVarDefinition(const VarDecl *D,
       if (D->getType()->isReferenceType())
         T = D->getType();
 
-      if (getLangOpts().CPlusPlus) {
+      if (getLangOpts().HLSL &&
+          getHLSLRuntime().isResourceRecordType(D->getType().getTypePtr())) {
```

Comment:
```suggestion
          D->getType().getTypePtr()->isHLSLResourceRecord()) {
```
Since we have `isHLSLResourceRecord()` on `clang::Type` we no longer need the `CGHLSLRuntime::isResourceRecordType()`, right?